components/ui/antiwork-footer.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+
+import { Shuffle } from "lucide-react";
+import { useCallback, useState } from "react";
+import { Button } from "@/components/ui/button";
+
+export default function AntiworkFooter() {
+  const [footerBgColor, setFooterBgColor] = useState("#000000");
+  const [footerTextColor, setFooterTextColor] = useState("#ffffff");
+
+  const generateRandomColors = useCallback(() => {
+    const generateRandomColor = () =>
+      `#${Math.floor(Math.random() * 16777215)
+        .toString(16)
+        .padStart(6, "0")}`;
+
+    const getContrastRatio = (color1: string, color2: string) => {
+      const luminance = (color: string) => {
+        const rgb = parseInt(color.slice(1), 16);
+        const r = (rgb >> 16) & 0xff;
+        const g = (rgb >> 8) & 0xff;
+        const b = (rgb >> 0) & 0xff;
+        return 0.2126 * r + 0.7152 * g + 0.0722 * b;
+      };
+      const l1 = luminance(color1);
+      const l2 = luminance(color2);
+      return (Math.max(l1, l2) + 0.05) / (Math.min(l1, l2) + 0.05);
+    };
+
+    let bgColor, textColor;
+    do {
+      bgColor = generateRandomColor();
+      textColor = generateRandomColor();
+    } while (getContrastRatio(bgColor, textColor) < 4.5);
+
+    setFooterBgColor(bgColor);
+    setFooterTextColor(textColor);
+  }, []);
+
+  return (
+    <footer
+      className="transition-colors duration-300"
+      style={{ backgroundColor: footerBgColor }}
+    >
+      <div className="container mx-auto flex h-[15vh] w-full items-center">
+        <div className="flex w-full items-center justify-between">
+          <div className="flex items-center">
+            <div className="flex flex-col items-start">
+              <a
+                href="https://antiwork.com/"
+                target="_blank"
+                rel="noopener noreferrer"
```

Comment:
The link is between AW's sites, I think we should not use `noreferrer`.
```suggestion
                rel="noopener"
```